src/Services/Catalog/Catalog.Presentation/Common/Middleware/ExceptionHandlingMiddleware.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Catalog.Application.Common.Exceptions;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Catalog.Presentation.Common.Middleware;
+
+public class ExceptionHandlingMiddleware
+{
+    private readonly RequestDelegate _requestDelegate;
+
+    public ExceptionHandlingMiddleware(RequestDelegate requestDelegate)
+    {
+        _requestDelegate = requestDelegate;
+    }
+
+    public async Task InvokeAsync(HttpContext context)
+    {
+        try
+        {
+            await _requestDelegate(context);
+        }
+        catch (ValidationException exception)
+        {
+            ProblemDetails problemDetails = new ()
+            {
+                Status = StatusCodes.Status400BadRequest,
+                Type = "https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.1",
+                Title = "Validation error",
+                Detail = "One or more validation errors has occurred",
+            };
+
+            if (exception.ValidationErrors is not null)
+            {
+                problemDetails.Extensions["errors"] = exception.ValidationErrors;
+            }
+
+            context.Response.StatusCode = StatusCodes.Status400BadRequest;
+
+            await context.Response.WriteAsJsonAsync(problemDetails);
+        }
+        catch (ExistingCategoryException exception)
+        {
+            ProblemDetails problemDetails = new ()
+            {
+                Status = StatusCodes.Status400BadRequest,
+                Type = "https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.1",
+                Title = "Existing category",
+                Detail = exception.Message,
+            };
+
+            context.Response.StatusCode = StatusCodes.Status400BadRequest;
+
+            await context.Response.WriteAsJsonAsync(problemDetails);
```

Comment:
context.Response.StatusCode = /*Code*/;
await context.Response.WriteAsJsonAsync(problemDetails).
is duplicated, try to generate  problemDetails and then write response